In [12]:
library(ggplot2)
options(repr.plot.height=4,repr.plot.width=6)

library(dplyr)

Cargar los datos en un dataframe llamado: airbnb

In [13]:
airbnb<-read.csv('data//airbnb.csv',sep = ',', stringsAsFactors = T)

Mostrar las primeras 6 filas del dataframe

In [14]:
head(airbnb,6)

Zipcode Neighbourhood.Cleansed Property.Type Room.Type       Accommodates
1 28004   Universidad            Apartment     Private room    2           
2 28004   Universidad            Apartment     Entire home/apt 6           
3 28004   Universidad            Apartment     Entire home/apt 3           
4 28004   Universidad            Loft          Entire home/apt 3           
5 28015   Universidad            Apartment     Entire home/apt 5           
6 28004   Universidad            Apartment     Entire home/apt 2           
  Bathrooms Bedrooms Beds Bed.Type
1 2         1        1    Real Bed
2 1         3        5    Real Bed
3 1         2        2    Real Bed
4 2         1        1    Real Bed
5 1         1        1    Real Bed
6 1         0        1    Real Bed
  Amenities                                                                                                                                                                                                                                                                                                                                                       
1 TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron                                                                                                                                                                             
2 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Indoor fireplace,Heating,Family/kid friendly,Washer,Dryer,Smoke detector,Carbon monoxide detector,Essentials,Shampoo                                                                                                                                                                                     
3 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Doorman,Heating,Family/kid friendly,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace                                                                                                                                                                                          
4 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Pets allowed,Breakfast,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace,translation missing: en.hosting_amenity_49,translation missing: en.hosting_amenity_50                                          
5 TV,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Pets allowed,Elevator in building,Heating,Family/kid friendly,Washer,Essentials,Iron                                                                                                                                                                                                              
6 TV,Cable TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Doorman,Elevator in building,Buzzer/wireless intercom,Heating,Washer,Smoke detector,Carbon monoxide detector,First aid kit,Safety card,Fire extinguisher,Essentials,Shampoo,24-hour check-in,Hangers,Hair dryer,Iron,Laptop friendly workspace,Self Check-In,Doorman Entry
  Square.Feet Price Review.Scores.Rating
1  NA         35     NA                 
2  NA         92     96                 
3  NA         55     91                 
4  NA         60    100                 
5 538         75     90                 
6  NA         55     95

Renombrar las columnas de la siguiente forma:

| Nombre original | Nuevo nombre |
| - | - |
| Zipcode | CodigoPostal |
| Neighbourhood.Cleansed | Barrio |
| Property.Type	| TipoPropiedad |
| Room.Type | TipoAlquiler |
| Accommodates | MaxOcupantes |
| Bathrooms	| NumBanyos |
| Bedrooms	| NumDormitorios |
| Beds	|  NumCamas |
| Bed.Type	| TipoCama |
| Amenities | Comodidades |
| Square.Feet | PiesCuadrados |
| Price	| Precio |
| Review.Scores.Rating | Puntuacion |

In [15]:
newnames<-c("CodigoPostal","Barrio","TipoPropiedad","TipoAlquiler","MaxOcupantes","NumBanyos",
"NumDormitorios","NumCamas","TipoCama","Comodidades","PiesCuadrados","Precio","Puntuacion")
colnames(airbnb) <- newnames

Crea una nueva columna llamada MetrosCuadrados a partir de la columna PiesCuadrados.

Ayuda: 1 pie cuadrado son 0,092903 metros cuadrdados

In [16]:
#airbnb

In [7]:
#airbnb %>% select(PiesCuadrados)

In [17]:
#airbnb <- airbnb %>%
#  mutate(MetrosCuadrados = PiesCuadrados*0,3048)

airbnb$MetrosCuadrados <- airbnb$PiesCuadrados * 0.453592

In [18]:
head(airbnb,6)

CodigoPostal Barrio      TipoPropiedad TipoAlquiler    MaxOcupantes NumBanyos
1 28004        Universidad Apartment     Private room    2            2        
2 28004        Universidad Apartment     Entire home/apt 6            1        
3 28004        Universidad Apartment     Entire home/apt 3            1        
4 28004        Universidad Loft          Entire home/apt 3            2        
5 28015        Universidad Apartment     Entire home/apt 5            1        
6 28004        Universidad Apartment     Entire home/apt 2            1        
  NumDormitorios NumCamas TipoCama
1 1              1        Real Bed
2 3              5        Real Bed
3 2              2        Real Bed
4 1              1        Real Bed
5 1              1        Real Bed
6 0              1        Real Bed
  Comodidades                                                                                                                                                                                                                                                                                                                                                     
1 TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron                                                                                                                                                                             
2 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Indoor fireplace,Heating,Family/kid friendly,Washer,Dryer,Smoke detector,Carbon monoxide detector,Essentials,Shampoo                                                                                                                                                                                     
3 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Doorman,Heating,Family/kid friendly,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace                                                                                                                                                                                          
4 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Pets allowed,Breakfast,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace,translation missing: en.hosting_amenity_49,translation missing: en.hosting_amenity_50                                          
5 TV,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Pets allowed,Elevator in building,Heating,Family/kid friendly,Washer,Essentials,Iron                                                                                                                                                                                                              
6 TV,Cable TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Doorman,Elevator in building,Buzzer/wireless intercom,Heating,Washer,Smoke detector,Carbon monoxide detector,First aid kit,Safety card,Fire extinguisher,Essentials,Shampoo,24-hour check-in,Hangers,Hair dryer,Iron,Laptop friendly workspace,Self Check-In,Doorman Entry
  PiesCuadrados Precio Puntuacion MetrosCuadrados
1  NA           35      NA              NA       
2  NA           92      96              NA       
3  NA           55      91              NA       
4  NA           60     100              NA       
5 538           75      90        244.0325       
6  NA           55      95              NA

Miremos el código postal. Es una variable con entradas erroneas.
Hay valores como '', '-'  y '28' que deberían ser considerados como NA.
Así mismo también debería ser NA todos los que no compiencen por 28, ya que estamos con códigos postales de Madrid

El código postal 28002, 28004 y 28051 tienen entradas repetidas. 
Por ejemplo las entradas 28002\n20882 deberían ir dnetro de 28002

El codigo 2804 debería ser 28004, 2805 deberia ser 28005 y 2815 juncto con 2815 debería ser 28015

Limpia los datos de la columna Codigo Postal

In [19]:
#?substring

In [20]:
#grep("28002\n20882", pattern="\n") == 1

In [21]:
#table(substring(airbnb$CodigoPostal,1,2))
#table(substring(airbnb$CodigoPostal,1,2) == 28)
#filter(airbnb, substring(airbnb$CodigoPostal,1,2) == 28)

In [23]:
#count(filter(airbnb, CodigoPostal == ""))

In [24]:
#filter(airbnb, CodigoPostal == "-")

In [25]:
#count(filter(airbnb, CodigoPostal == "28"))

In [26]:
#filter(airbnb, CodigoPostal == "28002\n28002")

In [27]:
#count(filter(airbnb, CodigoPostal == "28002"))

In [28]:
#count(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28002'))

In [29]:
#airbnb$CodigoPostal[airbnb$CodigoPostal==''] <- NA

In [43]:
#Miremos el código postal. Es una variable con entradas erroneas. 
#Hay valores como '', '-' y '28' que deberían ser considerados como NA. 
#Así mismo también debería ser NA todos los que no compiencen por 28, 
#ya que estamos con códigos postales de Madrid
airbnb$CodigoPostal[airbnb$CodigoPostal==''] <- NA
airbnb$CodigoPostal[airbnb$CodigoPostal=='-'] <- NA
airbnb$CodigoPostal[airbnb$CodigoPostal=='28'] <- NA
airbnb$CodigoPostal[substring(airbnb$CodigoPostal,1,2)!='28'] <- NA
#El código postal 28002, 28004 y 28051 tienen entradas repetidas. 
#Por ejemplo las entradas 28002\n20882 deberían ir dnetro de 28002
# Verificamos las entradas duplicadas
#filter(airbnb, CodigoPostal == '28002')[duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28002')),]
#filter(airbnb, CodigoPostal == '28004')[duplicated(filter(airbnb, CodigoPostal == '28004')),]
#filter(airbnb, CodigoPostal == '28051')[duplicated(filter(airbnb, CodigoPostal == '28051')),]
airbnb$CodigoPostal[substring(as.character(airbnb$CodigoPostal),1,5)=='28002'] <- '28002'
airbnb$CodigoPostal[substring(as.character(airbnb$CodigoPostal),1,5)=='28051'] <- '28051'
airbnb <- airbnb[!duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28002')),]
airbnb <- airbnb[!duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28004')),]
airbnb <- airbnb[!duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28051')),]
#El codigo 2804 debería ser 28004, 2805 deberia ser 28005 y 2815 juncto con 2815 debería ser 28015
airbnb$CodigoPostal[airbnb$CodigoPostal=='2804'] <- '28004'
airbnb$CodigoPostal[airbnb$CodigoPostal=='2805'] <- '28005'
airbnb$CodigoPostal[airbnb$CodigoPostal=='2815'] <- '28015'

#Limpia los datos de la columna Codigo Postal

airbnb <- airbnb[!duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28004')),]
airbnb <- airbnb[!duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28005')),]
airbnb <- airbnb[!duplicated(filter(airbnb, substring(as.character(airbnb$CodigoPostal),1,5) == '28015')),]

Una vez limpios los datos ¿Cuales son los códigos postales que tenemos? 

In [33]:
order(unique(airbnb$CodigoPostal), decreasing = TRUE)

[1] 60 56 50 63 37 62 44 34 22 64 59 58 17 20 61 23 29  7 55 42 49 35 53 30 21
[26] 47 28 52 16 43 46 54 39 33 10 41 36 24 26 57 48 45 31 25 38 40 27  2  6  3
[51]  5 19 14 11 18  9 13  4  1 32 15 51 12  8

In [42]:
unique(airbnb$CodigoPostal)

[1] 28004  28015  28013  28005  28012  28014  28045  <NA>   28007  28028 
[11] 28009  28001  28006  28010  28002  28034  28050  28008  28011  28049 
[21] 28038  28053  28047  28025  28019  28024  28016  28036  28046  28039 
[31] 28020  28003  28029  28054  28041  28026  28058  28018  28030  28017 
[41] 28027  28043  28033  28055  28021  28032  28037  28022  28042  28094 
[51] 280013 28035  28040  28031  28044  28105  28023  28051  28850  28048 
[61] 28056  28060  28052 
77 Levels:  - 20013 20126 2015 25008 27004 27013 28 28001 280013 ... Madrid 28004

In [44]:
filter(airbnb, CodigoPostal == '28004')[duplicated(filter(airbnb, CodigoPostal == '28004')),]

CodigoPostal Barrio      TipoPropiedad TipoAlquiler    MaxOcupantes
876  28004        Universidad Apartment     Private room    2           
1105 28004        Justicia    Apartment     Private room    1           
1484 28004        Justicia    Apartment     Entire home/apt 4           
1508 28004        Justicia    Apartment     Entire home/apt 6           
     NumBanyos NumDormitorios NumCamas TipoCama
876  2         1              1        Real Bed
1105 1         1              1        Real Bed
1484 1         1              2        Real Bed
1508 1         2              3        Real Bed
     Comodidades                                                                                                                                                                        
876  TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron
1105 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Heating,Washer,Essentials,Shampoo                                                                           
1484 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Elevator in building,Buzzer/wireless intercom,Heating,Family/kid friendly,Washer,Fire extinguisher,Essentials,Shampoo       
1508 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Elevator in building,Buzzer/wireless intercom,Heating,Family/kid friendly,Washer,Fire extinguisher,Essentials,Shampoo       
     PiesCuadrados Precio Puntuacion MetrosCuadrados
876  NA             35    NA         NA             
1105 NA             20    90         NA             
1484 NA            185    NA         NA             
1508 NA            195    NA         NA

¿Cuales son los 5 códigos postales con más entradas? ¿Y con menos? ¿Cuantas entradas tienen?

¿Cuales son los barrios que hay en el código postal 28012?

¿Cuantas entradas hay en cada uno de esos barrios para el codigo postal 28012?

¿Cuantos barrios hay en todo el dataset airbnb? ¿Cuales son?

¿Cuales son los 5 barrios que tienen mayor número entradas?

¿Cuantos Tipos de Alquiler diferentes hay? ¿Cuales son? ¿Cuantas entradas en el dataframe hay por cada tipo?

## Second section

Muestra el diagrama de cajas del precio para cada uno de los diferentes Tipos de Alquiler

Cual es el precio medio de alquiler de cada uno, la diferencia que hay ¿es estadísticamente significativa?
¿Con que test lo comprobarías?

Filtra el dataframe cuyos tipo de alquiler sea  'Entire home/apt' y guardalo en un dataframe llamado 
*airbnb_entire*.
Estas serán las entradas que tienen un alquiler del piso completo.

¿Cuales son los 5 barrios que tienen un mayor número de apartamentos enteros en alquiler?
Nota: Mirar solo en airbnb_entire. A partir de este punto y hasta que se diga lo contrario partiremos de airbnb_entire.

¿Cuales son los 5 barrios que tienen un mayor precio medio de alquiler para apartamentos enteros (es decir, del dataframe airbnb_entire)?

¿Cual es su precio medio?

Ayuda: Usa la función aggregate `aggregate(.~colname,df,mean,na.rm=TRUE)`

¿Cuantos apartamentos hay en cada uno de esos barrios?

Mostrar una dataframe con el nombre del barrio, el precio y el número de entradas.

Ayuda: Podeis crear un nuevo dataframe con las columnas "Barrio" y "Freq" que contenga el número de entradas en cada barrio y hacer un merge con el dataframe del punto anterior.

Partiendo del dataframe anterior, muestra los 5 barrios con mayor precio, pero que tengan más de 100 entradas de alquiler.

Dibuja el diagrama de densidad de distribución de los diferentes precios. Serían 5 gráficas, una por cada barrio.

Calcula el tamaño medio, en metros cuadrados, para los 5 barrios anteriores y muestralo en el mismo dataframe junto con el precio y número de entradas

Dibuja el diagrama de densidad de distribución de los diferentes tamaños de apartamentos. Serían 5 gráficas, una por cada barrio.

Esta claro que las medias de metros cuadrados de cada uno de estos 5 barrios parecen ser diferentes, pero ¿son estadísticamente diferentes?
¿Que test habría que usar para comprobarlo?

Para únicamente los pisos de alquiler en el barrio de Sol:

``barrio_sol<-subset(airbnb_entire,Barrio=="Sol")``

Calcular un modelo lineal que combine alguna de estas variables:
* NumBanyos
* NumDormitorios
* MaxOcupantes
* MetrosCuadrados

Primero calculamos la correlación para ver como se relacionan estas variables entre sí.

Se observa que la correlación entre el número de dormitorios y los metros cuadrados es sorprendentemente baja.
¿Son de fiar esos números?

Mediante un histograma o curvas de densidad podemos descartar números que notienen sentido en el dataframe barrio_sol,
para tener una matriz de correlación que tenga mayor sentido.


Una vez que hayamos filtrado los datos correspondientes calcular el valor o la combinación de valores que mejor nos permite obtener el precio de un inmueble.

¿Que variable es más fiable para conocer el precio de un inmueble, el número de habitaciones o los metros cuadrados?

Responde con su correspondiente margen de error del 95%, ¿cuantos euros incrementa el precio del alquiler por cada metro cuadrado extra del piso?

Responde con su correspondiente margen de error del 95%, ¿cuantos euros incrementa el precio del alquiler por cada habitación?

¿Cual es la probabilidad de encontrar, en el barrio de Sol, un apartamento en alquiler con 3 dormitorios?
¿Cual es el intervalo de confianza de esa probabilidad?